src/main/java/net/greghaines/jesque/worker/WorkerImpl.java

```diff
@@ -794,4 +800,32 @@ protected String lpoplpush(final String from, final String to) {
     public String toString() {
         return this.namespace + COLON + WORKER + COLON + this.name;
     }
+
+    class WorkerImplExceptionHandler implements ExceptionHandler {
+
+        ExceptionHandler defaultExceptionHandler = new DefaultExceptionHandler();
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RecoveryStrategy onException(final JobExecutor jobExecutor, final Exception ex,
+                                            final String curQueue) {
+            RecoveryStrategy recoveryStrategy;
+
+            if(ex instanceof JedisNoScriptException){
+                try {
+                    loadScripts();
+                    recoveryStrategy = RecoveryStrategy.RECONNECT;
+                } catch (IOException e) {
+                    recoveryStrategy = defaultExceptionHandler.onException(jobExecutor,ex,curQueue);;
+                }
+            } else {
+                recoveryStrategy = defaultExceptionHandler.onException(jobExecutor,ex,curQueue);;
+            }
+
+            return recoveryStrategy;
+        }
+    }
+
```

Comment:
Why did we touch the WorkerImpl class as part of this PR? Doesn't look related to what we're trying to achieve...

Reply:
Without reloading script after redis restart workers will not work.  And we are trying to achieve situation when listener will not only be called, but will have a chance to requeue aggregations :-)

Reply:
OK, cool :)